extensions/wikia/WallNotifications/WallNotificationsEveryone.class.php

```diff
@@ -130,23 +159,43 @@ public function getGlobalCacheBuster() {
 		return $val;
 	}
 
-	public function setEntityProcessed( $userId, $entityKey ) {
+	/**
+	 * Inserts a row to wall_notification_queue_processed table
+	 *
+	 * @param int $userId
+	 * @param string $entityKey in a form of "<revision ID>_<wiki ID>"
+	 */
+	private function markEntityAsProcessed($userId, $entityKey ) {
 		global $wgMemc;
 
 		wfProfileIn( __METHOD__ );
 
 		$cacheKey = $this->getEntityProcessedCacheKey( $userId, $entityKey );
 		$wgMemc->set( $cacheKey, true );
 
-		$this->getDB( true )->insert( 'wall_notification_queue_processed', [
-			'user_id' => $userId,
-			'entity_key' => $entityKey
-		], __METHOD__ );
+		// avoid duplicated entries for the same entity
+		//we use count(*) to check if there are entries in this table and that there's no need to push a row to wall_notifications
+		$this->getDB( true )->replace( self::WALL_NOTIFICATIONS_QUEUE_PROCESSED_TABLE, '',
+			[
+				'user_id' => $userId,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "user_id" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)